federation/cmd/federation-apiserver/app/server.go

```diff
@@ -212,8 +223,8 @@ func Run(s *options.ServerRunOptions) error {
 	installBatchAPIs(m, genericConfig.RESTOptionsGetter)
 	installAutoscalingAPIs(m, genericConfig.RESTOptionsGetter)
 
-	sharedInformers.Start(wait.NeverStop)
-	return m.PrepareRun().Run(wait.NeverStop)
+	sharedInformers.Start(stopCh)
+	return m.PrepareRun().NonBlockingRun(stopCh)
```

Comment:
If `m.PrepareRun().NonBlockingRun(stopCh)` fails, do we leak the sharedInformers? Shouldn't we start them only if `m.PrepareRun().NonBlockingRun(stopCh)` succeeds?